pubsub/google/cloud/pubsub/topic.py

```diff
@@ -499,3 +532,5 @@ def commit(self, client=None):
         message_ids = api.topic_publish(self.topic.full_name, self.messages[:])
         self.message_ids.extend(message_ids)
         del self.messages[:]
+        self._start_timestamp = time.time()
+        self._current_size = 0
```

Comment:
Seems a `reset_state` helper (or similar) would make sense here and in the constructor